spec/builtin-random.test.sh

```diff
@@ -0,0 +1,53 @@
+## oils_failures_allowed: 0
+## compare_shells: bash mksh zsh ash
+
+#### $RANDOM produces random numbers
+[[ -n $RANDOM ]] && echo set
+
+a=$RANDOM
+b=$RANDOM
+[[ $a -ne $b ]] && echo random
+
+## STDOUT:
+set
+random
+## END
+
+#### RANDOM=seed seeds a sequence of random numbers
```

Comment:
OK this is interesting.  It is a case where an assignment has a side effect

Similar to

- `PATH=` - clears the cache (I know you have a PR open for this)
- `LANG=` - calls setlocale()